source/distributed/shared_tria.cc

```diff
@@ -134,6 +135,19 @@ namespace parallel
       this->update_number_cache ();
     }
 
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::
+    copy_triangulation (const dealii::Triangulation<dim, spacedim> &other_tria)
+    {
+      Assert ((dynamic_cast<const dealii::parallel::distributed::Triangulation<dim,spacedim> *>(&other_tria) == NULL),
```

Comment:
i looks like you have forgotten to change this line according to the hidden comment https://github.com/dealii/dealii/pull/3061#discussion_r77512834